src/solc_0.8/common/BaseWithStorage/WithMetaTransaction.sol

```diff
@@ -46,44 +40,36 @@ contract WithMetaTransaction is WithAdmin {
     /// @param forwarder The address to query.
     /// @return whether or not forwarder is trusted.
     function isTrustedForwarder(address forwarder) public view returns (bool) {
-        return _metaTransactionContracts[forwarder] == METATX_2771;
+        return _metaTransactionProcessors[forwarder] == METATX_2771;
     }
 
     /// @dev Decide which sender address to use for this call.
     /// If the call came through our trusted forwarder, return the original sender.
     /// Otherwise, return `msg.sender`.
+    /// should be used in the contract anywhere instead of msg.sender !
     /// @return ret The sender of this call.
 
     function _msgSender() internal view virtual returns (address payable ret) {
         if (isTrustedForwarder(msg.sender)) {
-            return _forceMsgSender();
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                ret := shr(96, calldataload(sub(calldatasize(), 20)))
+            }
```

Comment:
Tentative: Rolled `_forceMsgSender()` into `_msgSender()` (As per https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.0.0-beta.0/contracts/metatx/ERC2771Context.sol`

Ideally I want to see if this would break any existing tests.